### PR TITLE
applications: nrf_desktop: Add BAS documentation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds BAS documentation.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@greg-fer, @pdunaj, could you review?

File: applications/nrf_desktop/doc/bas.rst [hunk 1]
```diff
@@ -3,4 +3,27 @@
...
-The ``bas`` module is used to control the GATT Battery Service.
+The ``bas`` module is used to define the GATT Battery Service and notify the subscribers about the battery level changes.
+
+Module Events
+*************
+
++---------------------+-------------------------+-------------+---------------+------------------+
+| Source Module       | Input Event             | This Module | Output Event  | Sink Module      |
++=====================+=========================+=============+===============+==================+
+| :ref:`battery_meas` | ``battery_level_event`` | ``bas``     |		      |			 |
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/bas.rst]:
```suggestion
| :ref:`battery_meas` | ``battery_level_event`` | ``bas``     |		|		 |
```
Make the vertical lines aligned in the table, please.

File: applications/nrf_desktop/doc/bas.rst [hunk 2]
```diff
@@ -3,4 +3,27 @@
...
++---------------------+-------------------------+-------------+---------------+------------------+
+
+Configuration
+*************
+
+The module is enabled with the ``CONFIG_DESKTOP_BAS_ENABLE`` option.
+The option is selected by ``CONFIG_DESKTOP_HID_PERIPHERAL`` - Battery Service is required for HID peripheral device.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/bas.rst]:
```suggestion
The option is selected by ``CONFIG_DESKTOP_HID_PERIPHERAL`` -- Battery Service is required for the HID peripheral device.
```

File: applications/nrf_desktop/doc/bas.rst [hunk 3]
```diff
@@ -3,4 +3,27 @@
...
+
+Implementation details
+**********************
+
+The module uses :c:macro:`BT_GATT_SERVICE_DEFINE` to define the GATT Service.
+On ``battery_level_event`` the module updates the battery level and notifies the subscribers.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/bas.rst]:
```suggestion
On ``battery_level_event``, the module updates the battery level and notifies the subscribers.
```

File: applications/nrf_desktop/doc/bas.rst [hunk 4]
```diff
@@ -3,4 +3,27 @@
...
+
+More detailed information regarding GATT are available in Zephyr documentation: :ref:`zephyr:bt_gatt`.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/bas.rst]:
```suggestion
More detailed information about GATT are available in the Zephyr documentation: :ref:`zephyr:bt_gatt`.
```

File: applications/nrf_desktop/doc/bas.rst [hunk 5]
```diff
@@ -3,4 +3,27 @@
 GATT Battery Service
 ####################
 
-The ``bas`` module is used to control the GATT Battery Service.
+The ``bas`` module is used to define the GATT Battery Service and notify the subscribers about the battery level changes.
```
Comment (greg-fer) [inline applications/nrf_desktop/doc/bas.rst]:
```suggestion
Use the ``bas`` module to enable and define the GATT Battery Service, and notify the subscribers about the battery level changes.
```

File: applications/nrf_desktop/doc/bas.rst (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/doc/bas.rst]: On my laptop the lines are aligned, I think its related to the way how Github displays it.

Comment (pdunaj) [reply applications/nrf_desktop/doc/bas.rst]: these are probably tabs

Comment (MarekPieta) [reply applications/nrf_desktop/doc/bas.rst]: Should we use spaces instead of tabs in `*.rst` files ?

**Comment by MarekPieta** [comment]
Switched from tabs to spaces in array alignment

**Pull request merged**